### PR TITLE
tf_cleanup

### PR DESCRIPTION
**Pull request by davidstanke**
Cleanups / refactors as recommended by @bleything (thank you!)

### CODE REVIEW

**Comment by davidstanke** [comment]
FYI, I was having continued auth issues invoking cloud build inside a terraform provisioner, so I pulled it out of TF and now it happens in the bash script before TF is run. (I may try to put it back in at some point, but for now this'll make it easier to progress with the rest of the terraforming).

**Comment by dinagraves** [comment]
thumbsup

**Comment by dinagraves** [comment]
> FYI, I was having continued auth issues invoking cloud build inside a terraform provisioner, so I pulled it out of TF and now it happens in the bash script before TF is run. (I may try to put it back in at some point, but for now this'll make it easier to progress with the rest of the terraforming).

Seems fine - There are going to have to be a few things that happen outside of the Terraform script, so that's fine.

**Review by dinagraves — APPROVED** [review]

File: experimental/terraform/setup.sh [hunk 1]
```diff
@@ -30,14 +30,22 @@ export BILLING_ACCOUNT=$(gcloud beta billing projects describe ${PARENT_PROJECT}
...
 gcloud beta billing projects link ${FOURKEYS_PROJECT} --billing-account=${BILLING_ACCOUNT}
+export PARENT_PROJECTNUM=$(gcloud projects describe ${PARENT_PROJECT} --format='value(projectNumber)')
 
 # FOR DEVELOPMENT ONLY: purge all TF state
+echo "Purging TF state [FOR DEVELOPMENT ONLY]"
 rm -rf .terraform terraform.tfstate* terraform.tfvars
 
+# build the event-handler container (using parent project) and stash it in the fourkeys project
+echo "building event-handler container"
+gcloud services enable cloudbuild.googleapis.com
```
Comment (bleything) [inline experimental/terraform/setup.sh:41]: do you want to specify `--project=${FOURKEYS_PROJECT}` here too? you're explicit about it in most other commands.

File: experimental/terraform/setup.sh [hunk 2]
```diff
@@ -30,14 +30,22 @@ export BILLING_ACCOUNT=$(gcloud beta billing projects describe ${PARENT_PROJECT}
...
+gcloud services enable containerregistry.googleapis.com --project=${FOURKEYS_PROJECT}
+gcloud projects add-iam-policy-binding ${FOURKEYS_PROJECT} --member="serviceAccount:${PARENT_PROJECTNUM}@cloudbuild.gserviceaccount.com" --role="roles/storage.admin"
```
Comment (bleything) [inline experimental/terraform/setup.sh:43]: what's this for?

**Review by bleything — APPROVED** [review]
Love it! Left a couple of comments about the bash script but the tf code looks great.

One thing I'd suggest if you haven't already is running `terraform fmt` in both the terraform directory and the module directory. If you're interested I have an Actions config that checks formatting.

File: experimental/terraform/setup.sh (hunk 1 above)
Comment (davidstanke) [reply experimental/terraform/setup.sh:41]: Good question! After much tinkering, I found that the most reliable solution was to build the container using the parent project, then push it to the registry in the child project. So it's not necessary to specify the projectID for Cloud Build, but it _is_ necessary for Container Registry. Perhaps I should make it explicit here though to clarify.

File: experimental/terraform/setup.sh (hunk 2 above)
Comment (davidstanke) [reply experimental/terraform/setup.sh:43]: Same as above -- to push a container from one project into another project, I need this permission. Thanks for the review!

**Comment by davidstanke** [comment]
> If you're interested I have an Actions config that checks formatting.

I'd definitely be interested... I've been manually running `terraform fmt -recursive` as I go but enforcing it makes sense as we scale this up.

**Pull request merged**